lambdas/data-migration2/src/executions.ts

```diff
@@ -62,16 +64,55 @@ export const migrateExecutionRecord = async (
   return cumulusId;
 };
 
+const processExecutionItems = async (
+  items: AWS.DynamoDB.DocumentClient.AttributeMap[],
+  migrationResult: MigrationResult,
+  knex: Knex,
+  loggingInterval: number,
+  writeConcurrency?: number
+) => {
+  const updatedResult = migrationResult;
+  await pMap(
+    items,
+    async (dynamoRecord) => {
+      updatedResult.total_dynamo_db_records += 1;
+
+      if (updatedResult.total_dynamo_db_records % loggingInterval === 0) {
+        logger.info(`Batch of ${loggingInterval} execution records processed, ${migrationResult.total_dynamo_db_records} total`);
```

Comment:
Not that it's a 'worth fixing' concern, but can this result in duplicate log entries depending on processing speed/concurrency?

Reply:
So far I haven't seen that. I agree that this approach to the batch logging is less than ideal but I couldn't see a way out of it without much more substantial refactoring. And this approach seems remarkably stable and to result in logs getting printed in the right order no matter the order of parallelism